NdefMessage.cpp

```diff
@@ -156,96 +200,118 @@ boolean NdefMessage::addRecord(NdefRecord& record)
     }
     else
     {
+#ifdef NDEF_USE_SERIAL
         Serial.println(F("WARNING: Too many records. Increase MAX_NDEF_RECORDS."));
+#endif
         return false;
     }
 }
 
-void NdefMessage::addMimeMediaRecord(String mimeType, String payload)
+void NdefMessage::addMimeMediaRecord(const char *mimeType, const char* payload)
 {
-
-    byte payloadBytes[payload.length() + 1];
-    payload.getBytes(payloadBytes, sizeof(payloadBytes));
-
-    addMimeMediaRecord(mimeType, payloadBytes, payload.length());
+	addMimeMediaRecord(mimeType, reinterpret_cast<const byte*>(payload), strlen(payload));
 }
 
-void NdefMessage::addMimeMediaRecord(String mimeType, uint8_t* payload, int payloadLength)
+void NdefMessage::addMimeMediaRecord(const char *mimeType, const byte* payload, uint16_t payloadLength)
 {
-    NdefRecord r = NdefRecord();
+    NdefRecord  r;
     r.setTnf(TNF_MIME_MEDIA);
 
-    byte type[mimeType.length() + 1];
-    mimeType.getBytes(type, sizeof(type));
-    r.setType(type, mimeType.length());
-
+    r.setType(reinterpret_cast<const byte*>(mimeType), strlen(mimeType));
     r.setPayload(payload, payloadLength);
+    addRecord(r);
+
+}
 
+void NdefMessage::addUnknownRecord(const byte *payload, int payloadLength)
+{
+    NdefRecord  r;
+    r.setTnf(TNF_UNKNOWN);
+
+    r.setType(payload, 0);
+    r.setPayload(payload, payloadLength);
     addRecord(r);
+
 }
 
-void NdefMessage::addTextRecord(String text)
+
+void NdefMessage::addTextRecord(const char *text)
 {
-    addTextRecord(text, "en");
+    addTextRecord(text, "");
```

Comment:
Can't remove the language code 'en', it's required for text records. Ideally it would default to en and allow users to override.

Reply:
I don't remember why I did this. Possibly because it was confusing me and I wanted a more manual way of writing the data, which looks like it should have been to call the 2-argument version of this function with the second argument empty. 

Restoring the default encoding here will probably require changes to a handful of calls in unit tests, i.e. from `addTextRecord(text)` to `addTextRecord(text, "")`